### PR TITLE
Moves preprocessor stuff to the gulpfile.

### PR DESCRIPTION
**Pull request by yurydelendik**
Moves all preprocessor logic and code from the make.js file and removes intermediate build/pdf.js, build/pdf.worker.js and build/viewer.js files (they were "last build configuration"-processed)

P.S. #7062

### CODE REVIEW

**Review by Snuffleupagus — APPROVED** [review]
When testing locally, everything still appears to work correctly.
r=me, with comments addressed.

Really nice work migrating even more tasks to gulp, thanks for doing this!

File: gulpfile.js [hunk 1]
```diff
@@ -296,6 +312,14 @@ function checkDir(path) {
   }
 }
 
+function getTempFile(prefix, suffix) {
+  mkdirp.sync(BUILD_DIR + 'tmp/');
```
Comment (yurydelendik) [inline gulpfile.js]: That is the tricky one: gulp tasks executed in parallel, so you may have multiple temp files opened. If we will try to count files of 'tmp/', there might be OS generated hidden files or just leftover after 'Ctrl+C' that might prevent removal of tmp.

Comment (Snuffleupagus) [inline gulpfile.js]: OK, let's just ignore that if it's not trivial; thanks for the explanation!

**Pull request merged**